pandas/core/reshape/reshape.py

```diff
@@ -725,7 +728,9 @@ def _convert_level_number(level_num: int, columns: Index):
     # time to ravel the values
     new_data = {}
     level_vals = mi_cols.levels[-1]
-    level_codes = sorted(set(mi_cols.codes[-1]))
+    level_codes = unique(mi_cols.codes[-1])
+    if sort:
+        level_codes = sorted(level_codes)
```

Comment:
this changes level_codes to a list?  can/should we preserve it as ndarray?

Reply:
Good idea. Used `np.sort` here instead